request.go

```diff
@@ -28,9 +29,27 @@ type (
 	// ExtraQueryValidationError the key used in `Context.Extra` to
 	// store the query validation errors.
 	ExtraQueryValidationError struct{}
+
+	// ExtraParseError the key used in `Context.Extra` to
+	// store specific request errors.
+	ExtraParseError struct{}
+)
+
+var (
+	// ErrInvalidQuery error when an invalid query string is passed.
+	ErrInvalidQuery = errors.New("parse.invalid-query")
+
+	// ErrInvalidJSONBody error when an empty or malformed JSON body is sent.
+	ErrInvalidJSONBody = errors.New("parse.json-invalid-body")
+
+	// ErrInvalidContentForType error when e.g. a multipart form is not actually multipart, or empty.
+	ErrInvalidContentForType = errors.New("parse.invalid-content-for-type")
+
+	// ErrErrorInRequestBody error when e.g. a incoming request is not received properly.
+	ErrErrorInRequestBody = errors.New("parse.error-in-request-body")
```

Comment:
If you have specific tweaking in mind, then I'm fine with merging it like this, since this feels more consistent with other translatable strings. Should be fine do have you do it afterwards to your liking?

Reply:
Yes don't change anything, I'll see what I do with them after this is merged. Thanks a lot.